pkg/service/cluster/service.go

```diff
@@ -568,20 +570,28 @@ func (s *Service) GetSession(ctx context.Context, clusterID uuid.UUID) (session
 		}
 	}
 
-	if ni.ClientEncryptionEnabled {
+	keyPair, err := s.loadTLSIdentity(clusterID)
+	if err != nil && !errors.Is(err, service.ErrNotFound) {
+		return session, err
+	}
+	if ni.ClientEncryptionEnabled && !errors.Is(err, service.ErrNotFound) {
```

Comment:
I think it would be clearer if you used:
```go
if ni.ClientEncryptionEnabled && err == nil {
```

Reply:
negation of NotFound in my view carries clearer intention of something being found, than err equal to nil.